crux-core/src/crux/api/v2/Database.java

```diff
@@ -0,0 +1,48 @@
+package crux.api.v2;
+
+import crux.api.ICruxAPI;
+import crux.api.ICruxDatasource;
+import clojure.lang.Keyword;
+
+import java.util.*;
+
+public class Database {
+    private final ICruxDatasource db;
+
+    private Database(ICruxDatasource db) {
+        this.db = db;
+    }
+
+    protected static Database database(ICruxAPI node) {
+        return new Database(node.db());
+    }
+
+    protected static Database database(ICruxAPI node, Date validTime) {
+        return new Database(node.db(validTime));
+    }
+
+    protected static Database database(ICruxAPI node, Date validTime, Date transactionTime) {
+        return new Database(node.db(validTime, transactionTime));
+    }
+
+    public Set<ResultTuple> query(Query query) {
+        Collection<List<?>> queryResult = db.q(query.toEdn());
+        List<LogicVar> symbols = query.findLogicVars();
+
+        Set<ResultTuple> resultSet = new HashSet<>();
+        for(List<?> result : queryResult){
+            resultSet.add(ResultTuple.resultTuple(symbols, result));
+        }
+
+        return resultSet;
+    }
+
+    public Collection<List<?>> query(String query) {
```

Comment:
It seems odd that return type is different. Should it also be `Set<ResultTuple>` or `Collection<ResultTuple>` instead?

Reply:
We were going to remove this function for now, in favor of making users use and construct query objects rather than just passing in a string.

Reply:
Fair enough, I'll wait for the merge of this branch into `master` and then review the whole package again, if you don't mind :-)

Reply:
Absolutely! There's a lot of great suggestions in here which I think we shall put in as soon as we commit our changes from earlier :)

Reply:
yep, agreed, removed for now